src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,98 @@
+package duke.command;
+
+import java.io.IOException;
+
+import duke.exception.DukeException;
+import duke.task.TaskList;
+import duke.util.Storage;
+import duke.util.Ui;
+
+/**
+ * DeleteCommand encapsulates the logic behind deleting a task from the TaskList.
+ */
+public class DeleteCommand extends Command {
+    private int taskNumber;
+
+    /**
+     * Constructor to create an instance of DeleteCommand.
+     *
+     * @param taskDetails a String representation containing details about the task
+     * @return an instance of DeleteCommand
+     */
+    public DeleteCommand(String taskDetails) {
+        this.taskNumber = Integer.parseInt(taskDetails);
+    }
+
+    /**
+     * Executes the DeleteCommand and removes the corresponding task from the TaskList.
+     * The updated TaskList is then stored into a txt file under data/tasks.txt
+     *
+     * @param tasks a list that keeps track of the tasks added/removed
+     * @param ui ui that handles the interaction with user inputs
+     * @param storage storage that handles the writing/reading of data from a txt file
+     * @throws DukeException if the TaskList is empty or an invalid task number in user's input
+     * @throws IOException if an error occurs while writing to a txt file
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException, IOException {
+        //Exception: Throw an error when user tries to delete from an empty list
+        if (tasks.isEmpty()) {
+            String errorMessage = "__________________________________________________\n"
+                    + "OOPS!!! There are no task left to be deleted!\n"
+                    + "__________________________________________________";
+            throw new DukeException(errorMessage);
+        }
+
+        //Exception: Throw an error when the second half after "delete" keyword is greater than task_list
+        if (taskNumber > tasks.size() || taskNumber < 1) {
```

Comment:
Why is there taskNumber < 1 condition in this line when it is already handled above?

Reply:
I'm assuming that taskNumber < 1 is the same check as isEmpty() check

Reply:
Hi! Sorry for the confusion, the isEmpty() check is to check if the tasks (TaskList) is empty while taskNumber here refers to the input from the user, in the event when the tasks (TaskList) contains task and they enter delete 0, delete -1 or any number that is < 1 we will throw an error since there can only be taskNumber starting from 1 onwards. Thank you for taking your time to review, I will edit the naming so that it is more intuitive.